# HG changeset patch
# User Andrew Janke <andrew@apjanke.net>
# Date 1531967648 14400
#      Wed Jul 18 22:34:08 2018 -0400
# Branch octfile-no-link-deps
# Node ID 6203a9fa669398b2f5087b0550634e77e18d7b90
# Parent  39fd627b2c5da4aa10d40be3b4f0f87d81814cca
Use -bundle -undefined dynamic_lookup for DLLs on macOS

diff --git configure.ac configure.ac
--- configure.ac
+++ configure.ac
@@ -650,25 +650,33 @@ AC_ARG_ENABLE([dl],
 
 if test $STATIC_LIBS = no && test $SHARED_LIBS = no; then
   AC_MSG_ERROR([You can't disable building both static AND shared libraries!])
 fi
 
 
 ### More configure argument checking related to linking
 
+case $host_os in
+  msdosmsvc | mingw* | AIX | BeOS | OS/2 )
+    DEFAULT_NO_UNDEFINED_LDFLAG="-no-undefined"
+  ;;
+  *)
+    DEFAULT_NO_UNDEFINED_LDFLAG=""
+  ;;
+esac
 AC_ARG_ENABLE([no-undefined],
   [AS_HELP_STRING([--disable-no-undefined],
     [don't pass -no-undefined to libtool when linking Octave and its shared libraries])],
   [case $enableval in
      yes) NO_UNDEFINED_LDFLAG="-no-undefined" ;;
      no)  NO_UNDEFINED_LDFLAG="" ;;
      *) AC_MSG_ERROR([bad value $enableval for --disable-no-undefined]) ;;
    esac],
-  [NO_UNDEFINED_LDFLAG="-no-undefined"])
+  [NO_UNDEFINED_LDFLAG="$DEFAULT_NO_UNDEFINED_LDFLAG"])
 AC_SUBST(NO_UNDEFINED_LDFLAG)
 
 AC_ARG_ENABLE([link-all-dependencies],
   [AS_HELP_STRING([--enable-link-all-dependencies],
     [link Octave and its shared libraries with all dependencies, not just those immediately referenced (should not be needed on most systems)])],
   [case $enableval in
      yes) link_all_deps=yes ;;
      no)  link_all_deps=no ;;
diff --git a/m4/acinclude.m4 b/m4/acinclude.m4
--- m4/acinclude.m4
+++ m4/acinclude.m4
@@ -2880,20 +2880,20 @@ AC_DEFUN_ONCE([OCTAVE_DEFINE_MKOCTFILE_D
     ;;
     alpha*-dec-osf*)
       CPICFLAG=
       CXXPICFLAG=
       FPICFLAG=
       SH_LDFLAGS="-shared -Wl,-expect_unresolved -Wl,'*'"
     ;;
     *-*-darwin*)
-      DL_LDFLAGS="-bundle -bundle_loader ${ac_top_build_prefix}libinterp/octave ${LDFLAGS}"
+      DL_LDFLAGS="-bundle -undefined dynamic_lookup -bind_at_load -bundle_loader ${ac_top_build_prefix}libinterp/octave ${LDFLAGS}"
       dnl Contains variables that are defined and undefined at this point, so use
       dnl appropriate quoting to defer expansion of ${bindir} and ${version}.
-      MKOCTFILE_DL_LDFLAGS='-bundle -bundle_loader ${bindir}/octave-${version}'"${EXEEXT}"
+      MKOCTFILE_DL_LDFLAGS='-bundle -undefined dynamic_lookup -bind_at_load -bundle_loader ${bindir}/octave-${version}'"${EXEEXT}"
       SH_LDFLAGS="-dynamiclib -single_module ${LDFLAGS}"
       case $canonical_host_type in
         powerpc-*)
           CXXPICFLAG=
           CPICFLAG=
           FPICFLAG=
         ;;
       esac
